broker-daemon/utils/base64-lex.spec.js

```diff
@@ -0,0 +1,102 @@
+const { expect } = require('test/test-helper')
+
+const encodeBase64Lex = require('./base64-lex')
+
+// The reason base64 doesn't preserve lexicographic order is because the index encoding
+// values for base64 do not match up with the index encoding values for ASCII characters.
+//
+// Base64 increases index using the pattern [A-Z][a-z][0-9][+][/]
+// ASCII increases index using the pattern [-][0-9][A-Z][_][a-z]
+describe('encodeBase64Lex', () => {
+  let base64LowerIndex
+  let base64HigherIndex
+  let lexLowerIndex
+  let lexHigherIndex
+
+  it('strips off base64 padding', () => {
+    // base64 encoding will have padding of only 1 or 2 '='
+    // (3 bytes are represented in 4 6-bit base64 digits, i.e. encoding is mod 3)
+    expect(encodeBase64Lex('12345=')).to.be.eql('pqrst')
+    expect(encodeBase64Lex('12345==')).to.be.eql('pqrst')
+  })
+
+  it('preserves lexicographic ordering for numbers vs uppercase letters', () => {
+    base64LowerIndex = 'A' // base64 index for 'A' is 0
+    base64HigherIndex = '0' // base64 index for '0' is 52
+
+    // ASCII character codes would incorrectly sort based on base64 for this case
+    expect(base64LowerIndex.charCodeAt(0)).to.be.greaterThan(base64HigherIndex.charCodeAt(0))
```

Comment:
I think this test would be better as:
1. two pieces of data, probably represented as Buffers or hex strings
2. use regular base64 and show they are misordered (`.toString('base64')`)
3. use the lex base64 and show they are correctly ordered.

Right now it exercises the logic, but not the code as much.